httpx/dispatch/http11.py

```diff
@@ -117,6 +117,8 @@ def __init__(
         Send a single `h11` event to the network, waiting for the data to
         drain before returning.
         """
+        if self.reader.is_connection_dropped():
+            raise NotConnected
```

Comment:
Checking the connection status here also prevents us from sending the request event which, [as @tomchristie mentioned](https://github.com/encode/httpx/issues/96#issuecomment-515011313), makes things ambiguous for non-idempotent requests.